### PR TITLE
open news in Default Browser

### PR DESCRIPTION
**Pull request by AuTiiTo**
Thanks to @gdfesta

### CODE REVIEW

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt [hunk 1]
```diff
@@ -11,7 +12,11 @@ import com.droidcba.kedditbysteps.commons.extensions.inflate
 import com.droidcba.kedditbysteps.commons.extensions.loadImg
 import kotlinx.android.synthetic.main.news_item.view.*
 
-class NewsDelegateAdapter : ViewTypeDelegateAdapter {
+class NewsDelegateAdapter(val mCommit: onViewSelectedListener) : ViewTypeDelegateAdapter {
```
Comment (juanchosaravia) [inline app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt]: Please can you rename this to something more specific to the listener? and remove the `m` prefix.
Like `viewActions`

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt [hunk 2]
```diff
@@ -32,6 +37,12 @@ class NewsDelegateAdapter : ViewTypeDelegateAdapter {
             author.text = item.author
             comments.text = "${item.numComments} comments"
             time.text = item.created.getFriendlyTime()
+            url.text = item.url
+            super.itemView.setOnClickListener(clickListener)
```
Comment (juanchosaravia) [inline app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt]: could it be this listener inlined?

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt [hunk 3]
```diff
@@ -22,16 +27,17 @@ class NewsDelegateAdapter : ViewTypeDelegateAdapter {
...
 
         fun bind(item: RedditNewsItem) = with(itemView) {
-            //Picasso.with(itemView.context).load(item.thumbnail).into(img_thumbnail)
             img_thumbnail.loadImg(item.thumbnail)
             description.text = item.title
             author.text = item.author
             comments.text = "${item.numComments} comments"
             time.text = item.created.getFriendlyTime()
+
+            super.itemView.setOnClickListener( {mCommit.onItemSelected(item.url)} )
```
Comment (juanchosaravia) [inline app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt]: As the last parameter is a function, you can remove the brackets in this way:
`super.itemView.setOnClickListener { mCommit.onItemSelected(item.url) }`

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt [hunk 4]
```diff
@@ -22,16 +27,17 @@ class NewsDelegateAdapter : ViewTypeDelegateAdapter {
         holder.bind(item as RedditNewsItem)
     }
 
-    class NewsViewHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
+    inner class NewsViewHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
             parent.inflate(R.layout.news_item)) {
 
         fun bind(item: RedditNewsItem) = with(itemView) {
-            //Picasso.with(itemView.context).load(item.thumbnail).into(img_thumbnail)
```
Comment (juanchosaravia) [inline app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt:29]: Please don't remove this comment as intentionally is there for tutorial purpose.

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt (hunk 1 above)
Comment (AuTiiTo) [reply app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt]: Ready

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt (hunk 4 above)
Comment (AuTiiTo) [reply app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt:29]: Sorry XD

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt (hunk 3 above)
Comment (AuTiiTo) [reply app/src/main/java/com/droidcba/kedditbysteps/features/news/adapter/NewsDelegateAdapter.kt]: Ready

**Pull request merged**